tests/integration/test_cogs_rates.py

```diff
@@ -21,55 +32,126 @@ def rates_cog(sample_bot):
 
 
 @pytest.mark.asyncio
-async def test_rates_compare_posted_rates(rates_cog, rates_url_1, rates_url_2):
-    output_path = rates_cog.output_paths[0]
+async def test_send_embed(rates_cog):
+    # Define an embed that matches what should be published by the cog
+    sample_embed = discord.Embed(
+        description="test",
+        title="ARK's Official server rates have just been updated!",
+        color=0x63BCC3,
+    )
+    sample_embed.set_image(url=EMBED_IMAGE)
 
-    # First request; no changes expected
-    rates_diff = await rates_cog.compare_posted_rates(rates_url_1, output_path)
-    assert len(rates_diff.items) == 0
+    # Publish the embed using the Rates cog
+    await rates_cog.send_embed(
+        description="test", url="www.mysite.com/dynamicconfig.ini"
+    )
 
-    # 2nd request; changes should be returned
-    rates_diff = await rates_cog.compare_posted_rates(rates_url_2, output_path)
+    # Ensure embed appears in the queue and nothing else.
+    assert dpytest.verify().message().embed(embed=sample_embed)
+    assert dpytest.verify().message().nothing()
 
-    assert len(rates_diff.items) == 3
-    assert (
-        RatesDiffItem(key="HarvestAmountMultiplier", old_val="3.0", new_val="2.0")
-        in rates_diff.items
-    )
-    assert (
-        RatesDiffItem(key="MatingIntervalMultiplier", old_val="0.6", new_val="0.7")
-        in rates_diff.items
-    )
-    assert (
-        RatesDiffItem(key="HexagonRewardMultiplier", old_val="1.5", new_val="2.5")
-        in rates_diff.items
+
+@pytest.mark.parametrize(
+    "sequential_handler",
+    [
+        (
+            SAMPLE_DYNAMICCONFIG_1,
+            SAMPLE_DYNAMICCONFIG_1,
+            SAMPLE_DYNAMICCONFIG_1,
+            SAMPLE_DYNAMICCONFIG_2,
+            SAMPLE_DYNAMICCONFIG_2,
+        )
+    ],
+    indirect=True,
+)
```

Comment:
The special thing here is the `indirect=True`. You can see above where the `sequential_handler` fixture requests the builtin pytest `request` fixture. When you use `indirect=True`, it sets values to the `request.param` field. This is how each test using that fixture can make use of different content (because we're injecting it at test run time, instead of test collection time).

The docs are pretty sparse on this, but it is documented in `pytest` docs.

Reply:
This is a really neat approach, glad to have learned it.